src/common/StackStringStream.h

```diff
@@ -86,7 +88,9 @@ class StackStringStream : public std::basic_ostream<char>
   ~StackStringStream() override = default;
 
   void clear() {
+    static std::ios_base::fmtflags const default_flags;
```

Comment:
but i think the application is responsible to reset the flag, isn't it?

Reply:
i still think that it's the application's responsibility to reset the flag. under most circumstances, we don't set the fmtflags when printing log messages, and it takes CPU cycles to reset the flags, do we really want to reset them for every single log line?

Reply:
Leaving it up to the application isn't really optimal in my opinion. We rely on fmtflags in most debug messages and any fmt changes in a prior message can arbitrarily affect others. I'm not really seeing a performance cost here since fmtflags is an int and it's a trivial assignment.

Reply:
@batrick in that case, we probably need to understand what `std::ios_base::fmtflags` is.

n4713 states that 

> The type fmtflags is a bitmask type (20.4.2.1.4).

and in libstdc++, it is defined like

```c++
  enum _Ios_Fmtflags
    {
...
    };
...
typedef _Ios_Fmtflags fmtflags;
```

so, there are two problem here

- to define a const variable we need to initialize it with some value
- what is the "some value"

so, probably, to reset it, we can just need to set to `0`. but `0` is not compliant to the C++17 standard. because the implementation of standard library can define the `fmtflags` as anything that supports bit operations. it can be a plain integer, enum or a full-blown class. so an alternative solution is to use `unsetf()`, like
```c++
unsetf(flags());
```
as `StackStringStream` inherits these methods from `std::basic_ostream<char>`, i think we can dispense with its parent class name.

Reply:
@tchaikov, yes I see. The old patch actually had a compile error for that reason.

I've taken a different approach of using a `static std::stringstream const` stream as the source of the default fmtflags. This should be more correct than `unsetf(flags())` as the default `fmtflags` may not actually be `0` -- i.e. there may be some flags set by default.